vpr/src/timing/timing_graph_builder.cpp

```diff
@@ -69,21 +265,27 @@ void TimingGraphBuilder::build(bool allow_dangling_combinational_nodes) {
         }
     }
 
+    //Walk through the netlist nets adding the edges representing each net to
+    //the timiing graph
     for (AtomNetId net : netlist_.nets()) {
         add_net_to_timing_graph(net);
     }
 
+    //Break any combinational loops (i.e. if the graph is not a DAG)
     fix_comb_loops();
 
+    //Levelize the graph for the timing analyzer
     tg_->levelize();
 }
 
+//Re-order the timing graph to be ordered in a traversal friendly manner
```

Comment:
I'd like a more detailed description how the optimization is done and why there is a need to remap ids.

Reply:
Done